plugins/woocommerce/client/admin/client/settings-payments/onboarding/providers/woopayments/components/onboarding/index.tsx

```diff
@@ -34,12 +35,12 @@ export default function WooPaymentsOnboarding( {
 	// Forces navigation to the current step only if the URL does not already match.
 	useEffect( () => {
 		if (
-			currentStep?.path &&
-			! location.pathname.endsWith( currentStep?.path ?? '' )
+			currentTopLevelStep &&
+			! location.pathname.endsWith( currentTopLevelStep?.path ?? '' )
 		) {
-			navigateToStep( currentStep.id );
+			navigateToStep( currentTopLevelStep.id );
 		}
-	}, [ currentStep, navigateToStep, location.pathname ] );
+	}, [ currentTopLevelStep, navigateToStep, location.pathname ] );
```

Comment:
I changed the URL structure to point only to top-level steps. However, there is nothing stopping us from changing it to link to backend sub-steps as well.